chorus_lib/src/core.rs

```diff
@@ -86,8 +96,8 @@ pub struct Unwrapper<L1: ChoreographyLocation> {
 
 impl<L1: ChoreographyLocation> Unwrapper<L1> {
     /// Takes the value located at the current location and returns its value
-    pub fn unwrap<V: ChoreographicValue>(&self, located: &Located<V, L1>) -> V {
-        located.value.clone().unwrap()
+    pub fn unwrap<V>(&self, located: Located<V, L1>) -> V {
+        located.value.unwrap()
```

Comment:
`unwrap` takes ownership of the located value. If the located value is accessed later, the programmer should explicitly `clone()`.